### PR TITLE
Update golang to 1.20.8

### LINKED ISSUE #1823: Update golang to 1.20
Go1.19 is no longer supported and we must update to go1.20 to receive security patches. Ref: [golang release policy](https://go.dev/doc/devel/release).

Updating is currently blocked on an [existing issue](https://github.com/golang/go/issues/61768) in the latest version of go1.20.

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

Resolves #1823 

## How this works

Updates all references to go1.20 and resolves the TODOs around updating to go1.20.

## How this was tested

CI

### CODE REVIEW

**Review by darioush — APPROVED** [review]

File: tests/fixture/testnet/local/network.go [hunk 1]
```diff
@@ -404,39 +404,17 @@ func (ln *LocalNetwork) GetURIs() []string {
...
-				NodeID:    node.NodeID,
-				StopError: err,
-			})
+			errs = append(errs, fmt.Errorf("failed to stop node %s: %w", node.NodeID, err))
 		}
 	}
 	if len(errs) > 0 {
-		// TODO(marun) When avalanchego updates to go 1.20, update to
-		// use the new capability to wrap multiple errors.
-		return &NetworkStopError{Errors: errs}
```
Comment (StephenButtolph) [inline tests/fixture/testnet/local/network.go:439]: @marun is this okay? or do you want to keep `NodeStopError`?

File: config/flags.go [hunk 2]
```diff
@@ -175,7 +175,6 @@ func addNodeFlags(fs *pflag.FlagSet) {
 	fs.Duration(BenchlistMinFailingDurationKey, constants.DefaultBenchlistMinFailingDuration, "Minimum amount of time messages to a peer must be failing before the peer is benched")
 
 	// Router
-	// TODO: Remove this flag in the future
```
Comment (danlaine) [inline config/flags.go:178]: Why's this TODO removed? Do we no longer intend to remove this flag?

**Review by danlaine — APPROVED** [review]

File: config/flags.go (hunk 2 above)
Comment (StephenButtolph) [reply config/flags.go:178]: The flag was already removed (but the todo wasn't)

**Review by abi87 — APPROVED** [review]

File: config/config.go [hunk 3]
```diff
@@ -950,7 +950,7 @@ func getAliases(v *viper.Viper, name string, contentKey string, fileKey string)
 
 	aliasMap := make(map[ids.ID][]string)
 	if err := json.Unmarshal(fileBytes, &aliasMap); err != nil {
-		return nil, fmt.Errorf("%w on %s: %s", errUnmarshalling, name, err)
+		return nil, fmt.Errorf("%w on %s: %w", errUnmarshalling, name, err)
```
Comment (hexfusion) [inline config/config.go:953]: :)

**Review by hexfusion — APPROVED** [review]

**Pull request merged**